gc/base/MemoryPoolAddressOrderedList.cpp

```diff
@@ -587,9 +610,52 @@ MM_MemoryPoolAddressOrderedList::collectorAllocate(MM_EnvironmentBase *env, MM_A
 	return addr;
 }
 
+/* Check the free entry being used and align it to appropriate boundaries for TLH allocation as required by GC.
+ * SATB Sweep has strict requirements for TLH to not span multiple Sweep Chunks. */
+bool
+MM_MemoryPoolAddressOrderedList::alignTLHForGC(MM_EnvironmentBase *env,  MM_HeapLinkedFreeHeader *freeEntry, uintptr_t *consumedSize)
+{
+	MM_GCExtensionsBase *extensions = env->getExtensions();
+
+	void *sweepChunkTopBoundary = NULL;
+	void *sweepChunkBaseBoundary = NULL;
+	void *tlhTopProjection = NULL;
+
+	if (NULL != _tlhAlignmentBoundary) {
+		Assert_MM_true(extensions->isSATBBarrierActive());
+
+		/* Determine the sweep chunk boundaries that contain the tlh being inited, we expect this chunk to be assigned during reassignChunks.*/
+		void* tlhBase = (void *)freeEntry;
+
+		Assert_MM_true(tlhBase >= _tlhAlignmentBoundary);
+
+		uintptr_t remainder = ((uintptr_t)tlhBase - (uintptr_t)_tlhAlignmentBoundary) % extensions->parSweepChunkSize;
+		sweepChunkBaseBoundary = (void *) ((uintptr_t) tlhBase - remainder);
+		/* sweepChunkTopBoundary may not be the real top boundary for the expected chunk, it's not wrapped around pool/region boundaries, which is fine. */
+		sweepChunkTopBoundary = (void *) ((uintptr_t) sweepChunkBaseBoundary + extensions->parSweepChunkSize);
+
+		Assert_MM_true(sweepChunkBaseBoundary <= tlhBase);
+		Assert_MM_true(sweepChunkTopBoundary > tlhBase);
+
+		tlhTopProjection = (void *) (((uint8_t *)freeEntry) + *consumedSize);
```

Comment:
make the casting for arithmetic consistent (either int8_t * or uintptr_t). 
look at the rest of the file what is used more

Reply:
well chunkBase and chunkTop could be uintptr_t (or int8_t *) to start with what would reduced amount of casting

Reply:
... but it's also nice to be consistent (if the rest of the code uses void *) so  this could too, so I don't have strong opinion what type of the points should be